### PR TITLE
Combine support for Decodable QuerySnapshot publisher

### PR DESCRIPTION
**Pull request by ziadtamim**
Here is a proposal for a `Decodable` publisher for the upstream `QuerySnapshot`.  

Here is a code snippet using the publisher returned by `getDocument()`

```swift
    let noBooks = [Book]()
    db.collection("books").getDocuments()
       .decode(type: Book.self)
       .collect()
       .replaceError(with: noBooks)
       .assign(to: \.books, on: self)
       .store(in: &cancellables)
```

### CODE REVIEW

File: Firestore/Swift/Source/Combine/Decodable+Combine.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (paulb777) [inline Firestore/Swift/Source/Combine/Decodable+Combine.swift:2]: 2021

**Comment by peterfriese** [comment]
Hi @ziadtamim -

sorry for closing this. We were expecting any in-flight PRs to automatically retarget to `master` once we merge `combine-main`.

Can you do this manually, and re-open the PR?

Thanks,
Peter

**Comment by paulb777** [comment]
I reopened and changed the base branch to master. Make sure you resync the local branch.

**Comment by ryanwilson** [comment]
Hi @ziadtamim - sorry for the long delay on this. I'm going through some old PRs and closing ones that we don't plan on merging anytime soon. It seems that patterns like async/await and Observable will be the future for async programming. Given this, I'm closing this PR. Please reach out if you'd like to re-open and discuss this PR more. Thanks for the contribution!

**Pull request closed without merging**